### PR TITLE
PR from dev to main for reviewing chargebyte board changes

### PR DESCRIPTION
**Pull request by M4GNV5**

### CODE REVIEW

File: chargebyte_board.py [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+        return self.parse_response(response)
+
+
+    def build_message(self, service_id: int, payload: bytearray)->bytearray:
+        if not isinstance(service_id,int):
+            raise TypeError
+        start_of_message = 0x02
+        length_of_message = 0x03 + len(payload)
+        device_adress = 0x00
+        response = bytearray([start_of_message,length_of_message,device_adress,service_id])+payload
```
Comment (M4GNV5) [inline chargebyte_board.py:93]: name this `header` not `response`?

File: chargebyte_board.py [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+
+    def join_bytes(self, low_byte:int, high_byte:int)->int:
+        return (high_byte << 8) | low_byte
+
+
+    def keep_service_alive():
+        pass
+
+
+    def test_device_one(self)->int|int|Enum:
```
Comment (M4GNV5) [inline chargebyte_board.py:137]: Using `|` means you return one of `int`, `int`, or `Enum`, but in reality you return all three of them as a tuple.
Also make sure to annotate the actual enum type not the generic `Enum`.

in this case use `tuple[int, int, ResetType]` as the type annotation

File: chargebyte_board.py [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+        response = self.send_packet(0x01, bytearray())
+        self.check_response_length(response,3)
+        software_version = response[0]
+        hardware_version = response[1]
+        last_reset_reason = ResetType(response[-1])
+        return software_version, hardware_version, last_reset_reason
+
+
+    def test_device_two(self)->int|Enum:
```
Comment (M4GNV5) [inline chargebyte_board.py:146]: same here: `tuple[int, ResetReason]`

File: chargebyte_board.py [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+
+
+    def read_response(self)->bytearray:
+        beginning = self.s.recv(1)
+        length = self.s.recv(1)
+        data = self.s.recv(int(length))
+        return bytearray([beginning,length])+data
+
+
+    def check_response(self, service_id:int, response:bytearray)->None:
```
Comment (M4GNV5) [inline chargebyte_board.py:104]: you can leave out `-> None` type annotations as this is the default

File: chargebyte_board.py [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+
+    def test_device_two(self)->int|Enum:
+        response = self.send_packet(0x04, bytearray())
+        self.check_response_length(response,3)
+        build = self.join_bytes(response[0],response[1])
+        last_reset_reason = ResetReason(response[2])
+        return build, last_reset_reason
+
+
+    def get_pwm(self)->int|int:
```
Comment (M4GNV5) [inline chargebyte_board.py:154]: `tuple[int, int]`

File: chargebyte_board.py [hunk 6]
```diff
@@ -0,0 +1,280 @@
...
+        response = self.send_packet(0x10, bytearray())
+        self.check_response_length(response,4)
+        frequency = self.join_bytes(response[0],response[1])
+        duty_cicle = self.join_bytes(response[2], response[3])
+        return frequency, duty_cicle
+
+
+    def set_pwm(self, frequency: int, dutycycle: int)->Enum:
```
Comment (M4GNV5) [inline chargebyte_board.py:162]: dont annotate `Enum` use `ControlPWM`

File: chargebyte_board.py [hunk 7]
```diff
@@ -0,0 +1,280 @@
...
+        low_freq = frequency & 0xff
+        high_freq = (frequency >> 8) & 0xff
+        low_duty = dutycycle & 0xff
+        high_duty =  (dutycycle >> 8) & 0xff
+        response = self.send_packet(0x11, bytearray([low_freq, high_freq, low_duty, high_duty]))
+        self.check_response_length(response,1)
+        return ControlPWM(response[0])
+
+
+    def control_pwm(self, control_code: int)->Enum:
```
Comment (M4GNV5) [inline chargebyte_board.py:172]: Consider using an enum type for the `control_code` parameter instead of `int`

**Review by M4GNV5 — COMMENTED** [review]
@giselaortt for each function check the return type (fix tuple typing, use specific Enum type) and check whether the parameters should be an enum as well (e.g. `command` argument given to `lock_unlock_cable_one`)

Other than that the code looks good. Keep up the great work!

File: chargebyte_board.py (hunk 2 above)
Comment (giselaortt) [reply chargebyte_board.py:137]: thank you, i didnt know that, i was given the wrong information. will fix it.

**Pull request merged**